### DISCUSSION TITLE
How do you get stock tuya firmware for reverse engineering?

### DISCUSSION THREAD

**Discussion by RealIndica** [ct-Open-Source/tuya-convert#973] [Q&A]
Wasn't too sure where to post this but this looks like the best place for it.

I'm looking to find where I can download stock tuya firmware images for reverse engineering purposes.

I am unable to find anywhere to do this and am unable to get firmware myself via a http proxy because all my devices are up to date on stock tuya firmware.

Thank you in advance

**Comment by kueblc** [comment]
[The wiki](https://github.com/ct-Open-Source/tuya-convert/wiki/Collaboration-document-for-PSK-Identity-02) has reverse engineering findings and data, including network captures and firmware samples.
